csharp/EmsApi.Client/V2/Wrappers/EmsApiRouteWrapper.cs

```diff
@@ -0,0 +1,135 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EmsApi.Client.V2.Wrappers
+{
+	/// <summary>
+	/// An API route wrapper is intended to convert the raw IEmsApi interface calls
+	/// to something that is a little more natural to work with inside .NET applications.
+	/// </summary>
+	public abstract class EmsApiRouteWrapper
+	{
+		/// <summary>
+		/// Creates a new instance of a route wrapper.
+		/// </summary>
+		/// <param name="api">
+		/// The raw API interface to make calls to.
+		/// </param>
+		public EmsApiRouteWrapper( IEmsApi api )
+		{
+			m_api = api;
+		}
+
+		/// <summary>
+		/// The reference to the raw api interface.
+		/// </summary>
+		/// <remarks>
+		/// This is private so classes which derive from this one must use the
+		/// <seealso cref="ContinueWithExceptionSafety{TRet}(Func{IEmsApi, Task{TRet}})"/>
+		/// method instead.
+		/// </remarks>
+		private IEmsApi m_api;
+
+		/// <summary>
+		/// Event that is fired whenever a low level API exception occurs.
+		/// </summary>
+		/// <remarks>
+		/// This is static so that this same event will fire for all derived wrapper classes.
+		/// </remarks>
+		internal static event EventHandler<ApiExceptionEventArgs> ApiMethodFailedEvent;
+
```

Comment:
I don't understand the rationale for static here. I think the idea is that you don't want event handlers to register to the wrong instance of the event. But it feels strange to me to make it static. Perhaps 'sealed' is a better fit?